docs/ref/class-based-views/base.txt

```diff
@@ -80,12 +80,19 @@ MRO is an acronym for Method Resolution Order.
     .. method:: setup(request, *args, **kwargs)
 
         Initializes view instance attributes: ``self.request``, ``self.args``,
-        and ``self.kwargs`` prior to :meth:`dispatch`.
+        and ``self.kwargs`` prior to :meth:`dispatch`. Also checks if both
+        ``get()`` and ``head()`` are defined. If ``get()`` is, but ``head()``
+        is not, then it sets::
+
+            self.head = self.get
 
         Overriding this method allows mixins to setup instance attributes for
         reuse in child classes. When overriding this method, you must call
         ``super()``.
 
+        This method also helps with unit testing class-based views. See
+        :ref:`unit-testing-class-based-views` for an example.
+
```

Comment:
I know we (I) merged this before but, I don't think we need to describe the implementation line for line. 

I'd be inclined to simplify this whole block to something such as: 

```suggestion:-13+4
        Performs key view initialization prior to :meth:`dispatch`.
        
        If overriding this method, you must call ``super()``.
        		          
```

(@felixxm What's your thought?)

Reply:
Agreed. I'm not even sure if we need to mention calling `super()` :thinking:

Reply:
No, that's important. (Because if you don't do it, your view won't work.)